LPAL-478 Add precommit hooks with instructions (#598)
ministryofjustice/opg-lpa: README.md

* Install `make` using the native package manager (assuming you are on Mac or Linux)
* Install [docker](https://docs.docker.com/get-docker/)
* Install [docker-compose](https://docs.docker.com/compose/install/)
* Install [homebrew](https://docs.brew.sh/) (mac only)

> ### clone repo

Author: ```suggestion
### Clone repo
```

cd opg-lpa
```

> ### install pre-commit hooks (mac)

Author: ```suggestion
### install pre-commit hooks (mac)
```

> ### install pre-commit hooks (mac)

Author: ```suggestion
### Install pre-commit hooks (mac)
```

> please install the precommit hooks as follows in the root of the repo directory (Mac only):

Author: ```suggestion
Install the precommit hooks as follows in the root of the repo directory (Mac only):
```

```bash
brew install php-code-sniffer
brew install php-cs-fixer
brew install phpstan
pre-commit install
```

> This will install the precommit hooks for the repo. this covers:

Author: ```suggestion
This will install the pre-commit hooks for the repo. This covers:
```

* PHP code fixers
* Terraform
* Secrets commit detection (AWS, general secrets)
* Whitespace and end of file fixers

Add more as needed to the `.pre-commit-config.yaml`.

> For linux users, revert to the instructions for installing phpcs, phpstan and precommit hooks:

Author: ```suggestion
For Linux users, revert to the instructions for installing phpcs, phpstan and pre-commit hooks:
```
